state/sshhostkeys.go

```diff
@@ -45,26 +46,57 @@ func (st *State) GetSSHHostKeys(tag names.MachineTag) (SSHHostKeys, error) {
 	return SSHHostKeys(doc.Keys), nil
 }
 
+// keysEqual checks if the ssh host keys are the same between two sets.
+// we shouldn't care about the order of the keys.
```

Comment:
Out of interest, why doesn't order matter in this case?

Reply:
The way I read this comment is that since they are sets element order should not matter for equality checks. Isn't this the case?

Reply:
I don't think SSH host keys care whether it is (A, B) or (B, A) they just care what values are there. That said, I couldn't be bothered, so I just went with the easy way of comparing them in order.
I guess we could just do a sort.Strings() before the comparison.

I know we had problems like that with IP addresses, because we were putting them in a map and then iterating a map. Which meant we kept getting InstancePoller thinking they had changed. I'll just put in the sort and be done with it.